homeassistant/components/lastfm/sensor.py

```diff
@@ -78,61 +77,69 @@ async def async_setup_entry(
 ) -> None:
     """Initialize the entries."""
 
-    lastfm_api = LastFMNetwork(api_key=entry.options[CONF_API_KEY])
+    coordinator: DataUpdateCoordinator[dict[str, LastFMUserData]] = hass.data[DOMAIN][
+        entry.entry_id
+    ]
     async_add_entities(
         (
-            LastFmSensor(lastfm_api.get_user(user), entry.entry_id)
-            for user in entry.options[CONF_USERS]
+            LastFmSensor(coordinator, username, entry.entry_id)
+            for username in entry.options[CONF_USERS]
         ),
         True,
     )
 
 
-class LastFmSensor(SensorEntity):
+class LastFmSensor(
+    CoordinatorEntity[DataUpdateCoordinator[dict[str, LastFMUserData]]], SensorEntity
+):
     """A class for the Last.fm account."""
 
     _attr_attribution = "Data provided by Last.fm"
     _attr_icon = "mdi:radio-fm"
 
-    def __init__(self, user: User, entry_id: str) -> None:
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator[dict[str, LastFMUserData]],
+        username: str,
+        entry_id: str,
+    ) -> None:
         """Initialize the sensor."""
-        self._user = user
-        self._attr_unique_id = hashlib.sha256(user.name.encode("utf-8")).hexdigest()
-        self._attr_name = user.name
+        super().__init__(coordinator)
+        self._username = username
+        self._attr_unique_id = hashlib.sha256(username.encode("utf-8")).hexdigest()
+        self._attr_name = username
         self._attr_device_info = DeviceInfo(
             configuration_url="https://www.last.fm",
             entry_type=DeviceEntryType.SERVICE,
             identifiers={(DOMAIN, f"{entry_id}_{self._attr_unique_id}")},
             manufacturer=DEFAULT_NAME,
-            name=f"{DEFAULT_NAME} {user.name}",
+            name=f"{DEFAULT_NAME} {username}",
         )
 
-    def update(self) -> None:
-        """Update device state."""
-        self._attr_native_value = STATE_NOT_SCROBBLING
-        try:
-            play_count = self._user.get_playcount()
-            self._attr_entity_picture = self._user.get_image()
-            now_playing = self._user.get_now_playing()
-            top_tracks = self._user.get_top_tracks(limit=1)
-            last_tracks = self._user.get_recent_tracks(limit=1)
-        except PyLastError as exc:
-            self._attr_available = False
-            LOGGER.error("Failed to load LastFM user `%s`: %r", self._user.name, exc)
-            return
-        self._attr_available = True
-        if now_playing:
-            self._attr_native_value = format_track(now_playing)
-        self._attr_extra_state_attributes = {
-            ATTR_PLAY_COUNT: play_count,
-            ATTR_LAST_PLAYED: None,
-            ATTR_TOP_PLAYED: None,
+    @property
+    def user(self) -> LastFMUserData | None:
```

Comment:
I think this should probably be renamed to `user_data` as it's not the user but the data from the user